### ISSUE TITLE
Returns all US timeszones?

### ISSUE THREAD

**Issue by renolc** [mmende/libphonenumber-geo-carrier#8]
I may be doing something wrong, but in all my tests so far, trying to determine the timezone of a US number seems to return _every_ US timezone (or at least a lot, most of which don't make sense).

Take for example the number of the Sacramento Zoo, +19168085888
```js
const number = parsePhoneNumber('+19168085888')
const tzs = await timezones(number)
/* tzs equals:
[
      'America/Adak',
      'America/Anchorage',
      'America/Anguilla',
      'America/Antigua',
      'America/Barbados',
      'America/Boise',
      'America/Cayman',
      'America/Chicago',
      'America/Denver',
      'America/Dominica',
      'America/Edmonton',
      'America/Fort_Nelson',
      'America/Grand_Turk',
      'America/Grenada',
      'America/Halifax',
      'America/Jamaica',
      'America/Juneau',
      'America/Los_Angeles',
      'America/Lower_Princes',
      'America/Montserrat',
      'America/Nassau',
      'America/New_York',
      'America/North_Dakota/Center',
      'America/Phoenix',
      'America/Port_of_Spain',
      'America/Puerto_Rico',
      'America/Regina',
      'America/Santo_Domingo',
      'America/St_Johns',
      'America/St_Kitts',
      'America/St_Lucia',
      'America/St_Thomas',
      'America/St_Vincent',
      'America/Toronto',
      'America/Tortola',
      'America/Vancouver',
      'America/Winnipeg',
      'Atlantic/Bermuda',
      'Pacific/Guam',
      'Pacific/Honolulu',
      'Pacific/Pago_Pago',
      'Pacific/Saipan'
] */
```

Am I doing something wrong here?

**Comment by renolc** [comment]
Looking into the code, I see you are pulling zones using only the country code, which would explain this. Are there any plans to enhance the method so it might return more granular results?

**Comment by mmende** [comment]
Hi @renolc 

The metadata used to get all the infos are extracted from the [libphonenumber repository](https://github.com/google/libphonenumber) directly which does not provide more granular results in terms of timezones unfortunately as these are only separated by the country calling code. If google decides to make the timezone metadata more granular some day, I will include it in the library of course.

**Comment by renolc** [comment]
@mmende
Hmm, looking at https://github.com/google/libphonenumber/blob/master/resources/timezones/map_data.txt, it does seem to go much more granular than the country code. Such as the test number I used above, the `map_data` file has `1916` map to `America/Los_Angeles`, which is the correct zone I would hope to receive.

EDIT: link to the specific line in the map file for reference: https://github.com/google/libphonenumber/blob/master/resources/timezones/map_data.txt#L1963

**Comment by mmende** [comment]
Sorry my fault. This was the same issue as with #7 and is fixed in the next version. In the upcoming version the library uses the longest match instead of only using the country calling code. I'll let you know when it is published. Thanks for the hint.

**Comment by mmende** [comment]
New version is published (v1.2.5)... Let me know if everything works fine now

**Comment by renolc** [comment]
Looks great! Thanks!
